### PR TITLE
Edit Content: Implement Control Value Accessor in the Binary Field

### LINKED ISSUE #26579: Edit Content: Implement Control Value Accessor in the Binary Field
### Parent
#25445

### User Story
As an editor, I want to be able to use the binary field component with the Angular forms as a field.

```[tasklist]
### Tasks
- [ ] The binary field must be able to be use in an Angular form
- [ ] Add the binary field field to the edit content form
- [ ] Binary field should send data to be saved
- [ ] Binary field should receive data and set the value
- [ ] [Field variables](https://www.dotcms.com/docs/latest/field-variables#BinaryField) should work as expected
```

### Proposed Objective
Static Content: "Core Features"

### Proposed Priority
Static Content: "Priority 3 - Average"

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
- Component is here: `core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts`

### Assumptions & Initiation Needs
- The Binary field field must be a component

### Quality Assurance Notes & Workarounds
N/A

### PR DESCRIPTION
**Pull request by rjvelazco**
### <samp>🤖 Generated by Copilot at 65827c9</samp>

Added support for binary fields in edit content mode. Refactored the `DotBinaryFieldComponent` to implement the `ControlValueAccessor` interface and communicate with the parent form. Updated the `DotEditContentFieldComponent` and the `DotEditContentFormComponent` to use the `DotBinaryFieldComponent` and pass the contentlet data to it. Added the `BINARY` value to the `FieldTypes` and `DotEditContentFieldTypes` enums and the field types components map. Added tests and mocks for the binary field type.

## Related Issue
Fixes #26579

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 65827c9</samp>

*  Implement the `ControlValueAccessor` interface for the `DotBinaryFieldComponent` to allow it to be used as a custom value accessor for the `FormControl` of the binary field ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554L16-R19), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554L64-R82), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554R88-R90), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554L82), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554L100-R122), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554R146-R159))
*  Remove the unused `tempFile` property from the `DotBinaryFieldComponent` class and use the `tempId` property instead to store and emit the temporary file id as the value of the binary field ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554L82))
*  Add a null check for the `field` input in the `getFieldVariables` method of the `DotBinaryFieldComponent` class to prevent errors if the input is not defined or initialized ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554L281-R316))
*  Make the overlay panel that shows the file metadata more responsive and adaptable to different screen sizes and orientations by replacing the `preview-metadata__container` class with the `preview-metadata__responsive` class in the `dot-binary-field-preview.component.html` template ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-1f4a9a7418368869300a97044d2d5333d4674f1b67425e31e58b98c8f8cd32e4L66-R66))
*  Add the `contentlet` input to the `DotEditContentFieldComponent` class and the `dot-edit-content-field.component.html` template to receive and pass the contentlet data to the child field components ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-f03434ad88316542a23c3fb0bf50b6650130632c3e7953d7ee3c882cb5950d06R38), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-685da2f87189fff825b7aec2666cfbabdb33c40f8f9d8c82df77ebb8652ccd4aR23), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-e1699e9269f6da395f7be873c698dcd539479427c31b61e1a6ea3c15de73beb9R8))
*  Add the `dot-binary-field` component as a case for the `fieldTypes.BINARY` switch in the `dot-edit-content-field.component.html` template and bind the relevant inputs to the parent component properties ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-685da2f87189fff825b7aec2666cfbabdb33c40f8f9d8c82df77ebb8652ccd4aR49-R55))
*  Add the `DotBinaryFieldComponent` to the testing module and the field types components map in the `dot-edit-content-field.component.ts` and `dot-edit-content-field.component.spec.ts` files ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-f03434ad88316542a23c3fb0bf50b6650130632c3e7953d7ee3c882cb5950d06L30-R32), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-23d2e2fd9b71e1819e10ae720cb79ecf5e4d75e51fee641b020017fe8186640fR8-R9), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-23d2e2fd9b71e1819e10ae720cb79ecf5e4d75e51fee641b020017fe8186640fL34-R37))
*  Add the `BINARY` value to the `FieldTypes` and `DotEditContentFieldTypes` enums in the `utils.ts` and `dot-edit-content-field.enum.ts` files to define the binary field type as a constant ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-72202882b348384879f6d696ea2c652a334c9cf4d2141b82ca531e37c8f3df2fL7-R8), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-aa72329fa1c18891b5d44042ac6ae09ff3d915621f16a084982f87d068c2453aL21-R22))
*  Add the `BINARY_FIELD_MOCK` object to the `mocks.ts` file and the `FIELDS_MOCK` array to create and include a mock binary field data for testing purposes ([link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-044e3c643ab55d05bd081afffe7251b7ebec1f4c383851aa1a219cdc626093f1R412-R435), [link](https://github.com/dotCMS/core/pull/26592/files?diff=unified&w=0#diff-044e3c643ab55d05bd081afffe7251b7ebec1f4c383851aa1a219cdc626093f1L428-R453))

#### Video

https://github.com/dotCMS/core/assets/72418962/0c7a1e4d-1548-44e6-95ac-5cca42a81f3d

### CODE REVIEW

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts [hunk 1]
```diff
@@ -351,3 +352,54 @@ describe('DotBinaryFieldComponent', () => {
...
+        component: MockFormComponent
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+    });
+
+    it('should create', () => {
+        expect(spectator.component).toBeTruthy();
+    });
```
Comment (zJaaal) [inline core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts:391]: Not needed

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts [hunk 2]
```diff
@@ -351,3 +352,54 @@ describe('DotBinaryFieldComponent', () => {
...
+            <dot-binary-field [field]="field" formControlName="binaryField"></dot-binary-field>
+        </form>
+    `
+})
+class MockFormComponent {
+    field = FIELD;
+    form = new FormGroup({
+        binaryField: new FormControl('')
+    });
+}
```
Comment (zJaaal) [inline core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts:377]: We have a mock for the form with the `FormGroupDirective` using that we don't need to create a component for testing.

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts:377]: That directive is in the `edit-content` library. I'll use it when we move the component to that library. For now, I'll add a comment.

File: dotCMS/build.gradle [hunk 3]
```diff
@@ -265,7 +265,7 @@ dependencies {
 
     starter group: 'com.dotcms', name: 'starter', version: 'empty_20230710', ext: 'zip'
     //Uncomment this line if you want to download the starter that comes with data
-    //starter group: 'com.dotcms', name: 'starter', version: '20230712', ext: 'zip'
+    starter group: 'com.dotcms', name: 'starter', version: '20230712', ext: 'zip'
```
Comment (oidacra) [inline dotCMS/build.gradle:268]: comment this please.

Comment (rjvelazco) [reply dotCMS/build.gradle]: Done!

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-form/dot-edit-content-form.component.html [hunk 4]
```diff
@@ -5,6 +5,7 @@
                 <dot-edit-content-field
                     *ngFor="let field of column.fields"
                     [field]="field"
+                    [contentlet]="formData.contentlet"
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/components/dot-edit-content-form/dot-edit-content-form.component.html:8]: Do we need test here?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts [hunk 5]
```diff
@@ -122,6 +143,20 @@ export class DotBinaryFieldComponent implements OnInit, AfterViewInit, OnDestroy
         this.cd.detectChanges();
     }
 
+    writeValue(): void {
+        /*
+            We can set a value here but we use the fields and contentlet to set the value
+        */
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts:149]: No, we don't, the value of the field is the `id`. In our case, we pass the `field` and `contentlet` to build the preview and only emit new values

[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts:149]: I'll remove the comment

**Review by fmontes — APPROVED** [review]

**Pull request merged**